packages/tm-core/src/modules/prompts/services/prompt-state-manager.ts

```diff
@@ -0,0 +1,284 @@
+ */
+
+import { getLogger } from '../../../common/logger/index.js';
+import { RuntimeStateManager } from '../../config/services/runtime-state-manager.service.js';
+import { PROMPT_STATE_VERSION } from '../constants.js';
+import type {
+	PromptAction,
+	PromptMetrics,
+	PromptState,
+	PromptStateStore,
+	TriggerType
+} from '../types.js';
+
+/**
+ * Key used for storing prompt state in RuntimeStateManager metadata
+ */
+const PROMPTS_METADATA_KEY = 'upgradePrompts';
+
+/**
+ * Manages prompt state using the existing RuntimeStateManager
+ */
+export class PromptStateManager {
+	private readonly logger = getLogger('PromptStateManager');
+	private readonly runtimeStateManager: RuntimeStateManager;
+	private cachedState: PromptStateStore | null = null;
+
+	constructor(projectRoot: string) {
+		this.runtimeStateManager = new RuntimeStateManager(projectRoot);
+	}
+
+	/**
+	 * Get the current prompt state
+	 */
+	async getState(): Promise<PromptStateStore> {
+		if (this.cachedState) {
+			return this.cachedState;
+		}
+
+		this.cachedState = await this.loadState();
+		return this.cachedState;
+	}
+
+	/**
+	 * Load state from RuntimeStateManager metadata
+	 */
+	private async loadState(): Promise<PromptStateStore> {
+		try {
+			await this.runtimeStateManager.loadState();
+			const runtimeState = this.runtimeStateManager.getState();
+			const promptsData = runtimeState.metadata?.[PROMPTS_METADATA_KEY];
+
+			if (promptsData && typeof promptsData === 'object') {
+				return this.validateAndMigrate(promptsData);
+			}
+		} catch (error) {
+			this.logger.warn('Failed to load prompt state, using defaults:', error);
+		}
+
+		return this.createDefaultState();
+	}
+
+	/**
+	 * Create default state
+	 */
+	private createDefaultState(): PromptStateStore {
+		return {
+			triggers: {},
+			metrics: {
+				totalTaskCount: 0,
+				tagCount: 0,
+				listCommandCount: 0,
+				tasksWithDependencies: 0
+			},
+			lastUpdated: new Date().toISOString(),
+			version: PROMPT_STATE_VERSION
+		};
+	}
+
+	/**
+	 * Validate and migrate state if needed
+	 */
+	private validateAndMigrate(data: any): PromptStateStore {
+		const state: PromptStateStore = {
+			triggers: data.triggers || {},
+			metrics: {
+				totalTaskCount: data.metrics?.totalTaskCount || 0,
+				tagCount: data.metrics?.tagCount || 0,
+				listCommandCount: data.metrics?.listCommandCount || 0,
+				tasksWithDependencies: data.metrics?.tasksWithDependencies || 0,
+				firstActivityAt: data.metrics?.firstActivityAt,
+				lastActivityAt: data.metrics?.lastActivityAt
+			},
+			lastUpdated: data.lastUpdated || new Date().toISOString(),
+			version: PROMPT_STATE_VERSION
+		};
+
+		return state;
+	}
+
+	/**
+	 * Save state to RuntimeStateManager metadata
+	 */
+	async saveState(): Promise<void> {
+		if (!this.cachedState) {
+			return;
+		}
+
+		try {
+			this.cachedState.lastUpdated = new Date().toISOString();
+			await this.runtimeStateManager.updateMetadata({
+				[PROMPTS_METADATA_KEY]: this.cachedState
+			});
```

Comment:
once you save state, wouldn't you want to flush cache and get it again so that your cachedState is always fresh ?